include/mips-emulator/executor.hpp

```diff
@@ -1,8 +1,19 @@
 #pragma once
 #include "mips-emulator/instruction.hpp"
+#include "memory.hpp"
+#include "register_file.hpp"
 
 namespace mips_emulator {
     namespace Executor {
+
+        // Declare non templated functions
+        static bool handle_rtype_instr(const Instruction instr,
+                                       RegisterFile& reg_file);
+        static bool handle_itype_instr(const Instruction instr,
+                                       RegisterFile& reg_file);
+        static bool handle_jtype_instr(const Instruction instr,
+                                       RegisterFile& reg_file);
+
         template <typename RegisterFile, typename Memory>
```

Comment:
We can remove the RegisterFile templating here as well and only have it be templated on the Memory type.